openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py

```diff
@@ -68,6 +68,43 @@ def fill_payment_id_and_statement_line_id_fields(env):
     )
 
 
+def fill_partial_reconcile_debit_and_credit_amounts(env):
+    # compute debit and credit amount when currencies are the same
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_partial_reconcile
+        SET debit_amount_currency = amount, credit_amount_currency = amount
+        WHERE debit_amount_currency IS NULL AND credit_amount_currency is NULL
+            AND credit_currency_id = debit_currency_id
+       """,
+    )
+    # compute debit and credit amount when currencies are different
+    partial_reconcile_lines = (
+        env["account.partial.reconcile"]
+        .search(
+            [
+                ("debit_amount_currency", "=", False),
+                ("credit_amount_currency", "=", False),
+            ]
+        )
+        .filtered(lambda line: line.credit_currency_id != line.debit_currency_id)
+    )
+    for line in partial_reconcile_lines:
+        line.debit_amount_currency = line.credit_currency_id._convert(
+            line.amount,
+            line.debit_currency_id,
+            line.company_id,
+            line.credit_move_id.date,
+        )
+        line.credit_amount_currency = line.debit_currency_id._convert(
+            line.amount,
+            line.credit_currency_id,
+            line.company_id,
+            line.debit_move_id.date,
+        )
```

Comment:
```suggestion
        line.debit_amount_currency = line.company_currency_id._convert(
            line.amount,
            line.debit_currency_id,
            line.company_id,
            line.debit_move_id.date,
        )
        line.credit_amount_currency = line.company_currency_id._convert(
            line.amount,
            line.credit_currency_id,
            line.company_id,
            line.credit_move_id.date,
        )
```

Reply:
Hi @MiquelRForgeFlow 

I can modify as you propose (I suppose this should be the same, except maybe for date ?), but I put it there on purpose since the lines that normally compute these fields are as follow (taken from [here](https://github.com/odoo/odoo/blob/077e554703b6d8a71c81fbf6783228d04e77d96f/addons/account/models/account_move.py#L4456)) :
```python
  min_debit_amount_residual_currency = credit_line.company_currency_id._convert(
      min_amount_residual,
      debit_line.currency_id,
      credit_line.company_id,
      credit_line.date,
  )
  min_credit_amount_residual_currency = debit_line.company_currency_id._convert(
      min_amount_residual,
      credit_line.currency_id,
      debit_line.company_id,
      debit_line.date,
  )
```
You can see there that there is on purpose a mix between debit and credit.
Let me know if you still prefer to go your way ?

Reply:
> I suppose this should be the same, except maybe for date ?

Yeah :)

Reply:
AFAIK line.amount is always in line.company_currency_id, thus that's why I think it should should be that way.

Reply:
Thanks, makes sense then !